core/planner.cpp

```diff
@@ -74,6 +74,17 @@ diveplan::~diveplan()
 {
 }
 
+diveplan diveplan::copy() const
```

Comment:
`copy()` seems to be creating the impression that this does actually a copy of the data structure (from a data point of view), and not a copy of the planned segments only. Should this rather be `copyPlannedSegments()`?

Reply:
You are right.

But the whole thing looks sketchy to me. Let's try something different.